src/disclosure-manager/_DisclosureContainer.jsx

```diff
@@ -34,6 +35,15 @@ const DisclosureContainer = injectIntl(({ intl, children, navigationPromptResolu
 
   const defaultPromptOptions = useMemo(() => getUnsavedChangesPromptOptions(intl), [intl]);
 
+  useEffect(() => {
+    const callback = disclosureManager.goBack || disclosureManager.closeDisclosure;
+    addCallback(callback);
+
+    return () => {
+      removeCallback(callback);
+    };
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Did we want to update this if disclosureManager changes?

Reply:
This is definitely worth discussing.  Re-running this if disclosure manager changes could introduce potential problems since we want to make sure that the array order is the disclosure order.  I was thinking that once disclosed these functions should never really change for the DisclosureContainer, but maybe that is a bad assumption.

One option that might make this more resilient would be to pass in a function here that references whatever disclosure manager is at the time of callback execution instead of passing in the disclosure manager functions directly.

Something like this:
```
const callback = () => { disclosureManager.goBack ? disclosureManager.goBack() : disclosureManager.closeDisclosure(); }
```

If we need to/want to support disclosureManager updates I think we would need some kind of "replaceCallback" function.  Any thoughts?

Reply:
The `disclosureManager` values are generated once for each disclosed component and stored in state. The value should not be changing.